sql/src/main/java/io/crate/rest/action/RestSQLAction.java

```diff
@@ -149,11 +154,44 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
 
     @VisibleForTesting
     User userFromRequest(RestRequest request) {
-        String user = request.header(AuthSettings.HTTP_HEADER_USER);
-        if (user == null) {
-            user = AuthSettings.AUTH_TRUST_HTTP_DEFAULT_HEADER.setting().get(settings);
+        String username = extractUsernameFromHttpBasicAuthHeader(
+            request.header(HttpHeaderNames.AUTHORIZATION.toString()));
+
+        // Fallback to deprecated setting
+        if (username == null) {
+            username = request.header(AuthSettings.HTTP_HEADER_USER);
+        }
+
+        // Fallback to trusted user from configuration
+        if (username == null) {
+            username = AuthSettings.AUTH_TRUST_HTTP_DEFAULT_HEADER.setting().get(settings);
+        }
+        return userManager.findUser(username);
+    }
+
+    private static String extractUsernameFromHttpBasicAuthHeader(String authHeaderValue) {
+        if (authHeaderValue == null) {
+            return null;
+        }
+        String[] creds = new String(
+            Base64.getDecoder().decode(authHeaderValue.replace("Basic ", "")),
+            StandardCharsets.UTF_8).split(":");
+        return creds[0];
+    }
+
+    public static Tuple<String, SecureString> extractCredentialsFromHttpBasicAuthHeader(String authHeaderValue) {
```

Comment:
imho this kind of functionality doesn't really belong into `RestSQLAction` - it's not only for `/_sql`